da-node/src/node_api/server.rs

```diff
@@ -0,0 +1,374 @@
+            size: blob.size,
+            owner_id: blob.owner_id.to_string(),
+            public_key: blob.public_key,
+            metadata: blob.metadata,
+            created_at: blob.created_at.to_rfc3339(),
+            deleted_at: blob.deleted_at.map(|dt| dt.to_rfc3339()),
+        };
+
+        let proto_blob = if request.include_content {
+            crate::node_api::proto::Blob {
+                metadata: Some(proto_metadata),
+                content: blob.content,
+            }
+        } else {
+            crate::node_api::proto::Blob {
+                metadata: Some(proto_metadata),
+                content: Vec::new(),
+            }
+        };
+
+        self.mark_blob_status_as_completed(Utc::now(), &blob.id, &request.peer_id)
+            .await?;
+
+        Ok(SyncResponse {
+            status: SyncStatus::Success as i32,
+            message: "Blob fetched successfully".to_string(),
+            response_data: Some(crate::node_api::proto::sync_response::ResponseData::Fetch(
+                FetchBlobResponse {
+                    blob: Some(proto_blob),
+                },
+            )),
+        })
+    }
+
+    #[tracing::instrument(name = "Handling delete blob request", skip(self))]
+    async fn handle_delete(&self, request: DeleteBlobRequest) -> Result<SyncResponse, Status> {
+        let blob_exists = self.does_blob_exist(&request.blob_id).await?;
+
+        if !blob_exists {
+            return Ok(SyncResponse {
+                status: SyncStatus::NotFound as i32,
+                message: format!("Blob not found: {}", request.blob_id),
+                response_data: Some(crate::node_api::proto::sync_response::ResponseData::Delete(
+                    DeleteBlobResponse { deleted: false },
+                )),
+            });
+        }
+
+        self.mark_blob_as_deleted(&request.blob_id).await?;
+
+        Ok(SyncResponse {
+            status: SyncStatus::Success as i32,
+            message: "Blob deleted successfully".to_string(),
+            response_data: Some(crate::node_api::proto::sync_response::ResponseData::Delete(
+                DeleteBlobResponse { deleted: true },
+            )),
+        })
+    }
+
+    #[tracing::instrument(name = "Handling sync status request", skip(self))]
+    async fn handle_sync_status(&self, request: SyncStatusRequest) -> Result<SyncResponse, Status> {
+        let statuses = sqlx::query_as::<
+            _,
+            (
+                String,
+                String,
+                String,
+                Option<DateTime<Utc>>,
+                Option<DateTime<Utc>>,
+            ),
+        >(
+            "SELECT peer_node_id, blob_id, sync_status, last_sync_attempt, last_successful_sync
+             FROM sync_status WHERE blob_id = ?",
+        )
+        .bind(&request.blob_id)
+        .fetch_all(&self.db_pool)
+        .await
+        .map_err(|e| Status::internal(format!("Database error: {}", e)))?;
```

Comment:
Not sure, that other nodes want to receive internal node's errors about failed database connection)

I suggest logging it instead and return empty message or only "internal database error"

Reply:
The same about other places where you do that